src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
+
+### Orleans.EventSourcing.**LogStorage**.LogConsistencyProvider
+
+This provider stores *the complete event sequence as a single object*, using a standard storage provider that can be configured independently.
+
+The data that is kept in storage is an object that contains a `List<EventType> object`, and some meta-data (a special tag that is used to avoid duplication of events when storage accesses fail).
+
+This provider does support `RetrieveConfirmedEvents`. All events are always available and kept in memory.
```

Comment:
Yes, this provider is not a great one. It is very similar to the initial draft of event sourcing in Orleans, where the grain state was defined as List<Event>.

The entire List<Event> is kept in memory because every time we read/write to/from storage, we have to read/write the entire list. 

 I think I say somewhere not to use this provider in production unless the number of events remains very small.